docs/content/concepts/cryptography/nautilus/using-nautilus.mdx

```diff
@@ -0,0 +1,338 @@
+---
+title: Using Nautilus
+description: Details on how to use Nautilus using the provided reproducible build template.
+---
+
+The [Nautilus framework](https://github.com/MystenLabs/nautilus) helps you deploy an AWS Nitro Enclave with all the necessary scaffolding, such as reproducible builds, signature formatting, and HTTPS traffic forwarding, so you can focus on implementing the off-chain computation logic inside the enclave.
+
+In addition, the framework provides an on-chain template that includes the minimal smart contract code required to register a Nautilus instance and its public key. These are the steps you take to use Nautilus in your dApp:
+
+1. Implement the enclave in Rust with the desired computation logic.
+2. Deploy a Move smart contract that stores the expected platform configuration registers (PCRs) and allows updates by the contract deployer.
+3. Deploy the enclave instance on AWS and register it on chain using its attestation document.
+4. Upload signed responses from the registered enclave, verify them on chain, and consume the results in your smart contract.
+
+## Purpose of this guide
+
+This guide walks you through the following steps:
+
+1. Writing and deploying a basic Nautilus off-chain instance using AWS Nitro Enclaves. The example instance runs a server that fetches weather data for a specific location.
+2. Writing a Move smart contract that registers the enclave by verifying its attestation and public key, then verifies the Nautilus response (signature and payload) on chain and mints an NFT containing the location and temperature data.
+
+The setup script performs the following actions:
+
+- Launches a preconfigured EC2 instance and allocates a Nitro Enclave.
+- Builds the Rust-based template application into an enclave image format (EIF) binary and runs it inside the enclave.
+- Configures required HTTP domains so the enclave can access external APIs via the parent EC2 instance (as the enclave itself has no internet access).
+- Exposes three endpoints to allow client-side communication with the enclave.
+
+When the enclave starts, it generates a fresh enclave key pair and exposes the following three endpoints:
```

Comment:
I changed this to read three instead of two. Correct?

Reply:
oh right, sorry and thanks.. these were originally two, but then we added the third later, but didn't update that number :)